### DISCUSSION TITLE
Why was `<ResizableContainerObserver />` deprecated and what is the recommended replacement?

### DISCUSSION THREAD

**Discussion by anmolshres98** [iTwin/appui#1139] [Q&A] [answered]
I am curious as to rationale behind why [\<ResizableContainerObserver />](https://github.com/iTwin/appui/blob/a31a180640d9c992b51df1c089359a233a4945f2/ui/core-react/src/core-react/utils/hooks/useResizeObserver.tsx#L277) was deprecated?

Is there a recommended third-party replacement or do you recommend apps to simply copy the now-deprecated implementation and maintain it themselves instead?

**Comment by anmolshres98** [comment]
@williamzqr fyi

**Comment by GerardasB** [comment, marked as answer]
We are cleaning up our API surface by deprecating low level components and utilities. For the most part, components/hooks like these are simply an implementation detail and it's better not to expose them. You could simply use https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver directly or one of npm packages (do your own research if you go this route) i.e.:
https://www.npmjs.com/package/@react-hook/resize-observer
https://www.npmjs.com/package/use-resize-observer
